fhirclient/models/fhirsearch.py

```diff
@@ -140,6 +165,23 @@ def perform_resources(self, server):
             
         return resources
 
+    # Use forward references to avoid circular imports
+    def perform_resources_iter(self, server) -> Iterator['Resource']:
+        """ Performs the search by calling `perform`, then extracts all Bundle
+        entries and returns an iterator of Resource instances.
+
+        :param server: The server against which to perform the search
+        :returns: An iterator of Resource instances
+        """
+        first_bundle = self.perform(server)
+
+        if not first_bundle or not first_bundle.entry:
+            return iter([])
```

Comment:
This check isn't needed, is it? `perform` can't return `None` and the `not .entry` check is handled by `__iter__`.

This method could be written as the following, I think?
```
for bundle in self.perform_iter(server):
  for entry in bundle:
    yield entry.resource
```